### PR TITLE
Add table_aws_cloudfront_distribution. Closes #109

### LINKED ISSUE #109: Add table_aws_cloudfront_distribution

### PR DESCRIPTION
**Pull request by visiit**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
vishalkumar@Vishals-MacBook-Air aws-test % TURBOT_PROFILE=shaktiman TF_VAR_aws_profile=shaktiman ./tint.js tests/aws_cloudfront_distribution
No env file present for the current environment:  staging
 Falling back to .env config
No env file present for the current environment:  staging
customEnv TURBOT_TEST_EXPECTED_TIMEOUT undefined
SETUP: tests/aws_cloudfront_distribution []
PRETEST: tests/aws_cloudfront_distribution
TEST: tests/aws_cloudfront_distribution
Running terraform
data.aws_region.alternate: Refreshing state...
data.aws_region.primary: Refreshing state...
data.aws_partition.current: Refreshing state...
data.aws_caller_identity.current: Refreshing state...
data.null_data_source.resource: Refreshing state...
aws_s3_bucket.test_bucket: Creating...
aws_s3_bucket.test_bucket: Still creating... [10s elapsed]
aws_s3_bucket.test_bucket: Creation complete after 19s [id=turbottest33626]
aws_cloudfront_distribution.named_test_resource: Creating...
aws_cloudfront_distribution.named_test_resource: Still creating... [10s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [20s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [30s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [40s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [50s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [1m0s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [1m10s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [1m20s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [1m30s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [1m40s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [1m50s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [2m0s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [2m10s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [2m20s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [2m30s elapsed]
aws_cloudfront_distribution.named_test_resource: Still creating... [2m40s elapsed]
aws_cloudfront_distribution.named_test_resource: Creation complete after 2m45s [id=E2OZ0DK2SSBK0W]
Warning: Deprecated Resource
The null_data_source was historically used to construct intermediate values to
re-use elsewhere in configuration, the same can now be achieved using locals
Warning: Interpolation-only expressions are deprecated
  on variables.tf line 57, in resource "aws_cloudfront_distribution" "named_test_resource":
  57:     domain_name = "${aws_s3_bucket.test_bucket.bucket_regional_domain_name}"
Terraform 0.11 and earlier required all non-constant expressions to be
provided via interpolation syntax, but this pattern is now deprecated. To
silence this warning, remove the "${ sequence from the start and the }"
sequence from the end of this expression, leaving just the inner expression.
Template interpolation syntax is still used to construct strings from
expressions when the template includes multiple interpolation sequences or a
mixture of literal strings and interpolations. This deprecation applies only
to templates that consist entirely of a single interpolation sequence.
Apply complete! Resources: 2 added, 0 changed, 0 destroyed.
Outputs:
domain_name = d3htj6iaslww58.cloudfront.net
etag = E19Y8YYYUB2JVE
is_ipv6_enabled = false
resource_aka = arn:aws:cloudfront::986325076436:distribution/E2OZ0DK2SSBK0W
resource_id = E2OZ0DK2SSBK0W
resource_name = turbottest33626
Running SQL query: test-get-query.sql
[
  {
    "akas": [
      "arn:aws:cloudfront::986325076436:distribution/E2OZ0DK2SSBK0W"
    ],
    "default_root_object": "index.html",
    "domain_name": "d3htj6iaslww58.cloudfront.net",
    "e_tag": "E19Y8YYYUB2JVE",
    "id": "E2OZ0DK2SSBK0W",
    "price_class": "PriceClass_200",
    "status": "Deployed",
    "tags": {
      "name": "turbottest33626"
    }
  }
]
✔ PASSED
Running SQL query: test-list-query.sql
[
  {
    "akas": [
      "arn:aws:cloudfront::986325076436:distribution/E2OZ0DK2SSBK0W"
    ],
    "comment": "",
    "domain_name": "d3htj6iaslww58.cloudfront.net",
    "e_tag": "E19Y8YYYUB2JVE",
    "enabled": false,
    "id": "E2OZ0DK2SSBK0W",
    "in_progress_invalidation_batches": 0,
    "is_ipv6_enabled": false
  }
]
✔ PASSED
Running SQL query: test-not-found-query.sql
null
✔ PASSED
Running SQL query: test-turbot-query.sql
[
  {
    "akas": [
      "arn:aws:cloudfront::986325076436:distribution/E2OZ0DK2SSBK0W"
    ],
    "region": "global",
    "tags": {
      "name": "turbottest33626"
    }
  }
]
✔ PASSED
POSTTEST: tests/aws_cloudfront_distribution
TEARDOWN: tests/aws_cloudfront_distribution
SUMMARY:
1/1 passed.
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
- Basic info

select
  id,
  arn,
  status,
  domain_name,
  enabled,
  e_tag,
  http_version,
  is_ipv6_enabled

from
  aws_cloudfront_distribution;

+----------------+--------------------------------------------------------------+----------+-------------------------------+---------+----------------+--------------+-----------------+
| id             | arn                                                          | status   | domain_name                   | enabled | e_tag          | http_version | is_ipv6_enabled |
+----------------+--------------------------------------------------------------+----------+-------------------------------+---------+----------------+--------------+-----------------+
| E3FW6WWZU7BYN5 | arn:aws:cloudfront::986325076436:distribution/E3FW6WWZU7BYN5 | Deployed | d2znvjafy01am9.cloudfront.net | false   | E2SW4ABH5VVI5V | HTTP2        | false           |
+----------------+--------------------------------------------------------------+----------+-------------------------------+---------+----------------+--------------+-----------------+

- List distributions with logging not enabled

+----------------+--------+----------------+-----------------+
| id             | bucket | logging_enable | include_cookies |
+----------------+--------+----------------+-----------------+
| E3FW6WWZU7BYN5 |        | false          | false           |
+----------------+--------+----------------+-----------------+

- List distributions with IPv6 DNS requests not enabled

select
  id,
  arn,
  status,
  is_ipv6_enabled
from
  aws_cloudfront_distribution
where
  is_ipv6_enabled = 'false';

+----------------+--------------------------------------------------------------+----------+-----------------+
| id             | arn                                                          | status   | is_ipv6_enabled |
+----------------+--------------------------------------------------------------+----------+-----------------+
| E3FW6WWZU7BYN5 | arn:aws:cloudfront::986325076436:distribution/E3FW6WWZU7BYN5 | Deployed | false           |
+----------------+--------------------------------------------------------------+----------+-----------------+

-  CloudFront distributions enforce field-level encryption

select
  id,
  arn,
  default_cache_behavior ->> 'FieldLevelEncryptionId' as field_level_encryption_id,
  default_cache_behavior ->> 'DefaultTTL' as default_ttl
from
  aws_cloudfront_distribution
WHERE
  default_cache_behavior ->> 'FieldLevelEncryptionId' <> '';

+----------------+--------------------------------------------------------------+---------------------------+-------------+
| id             | arn                                                          | field_level_encryption_id | default_ttl |
+----------------+--------------------------------------------------------------+---------------------------+-------------+
| E3FW6WWZU7BYN5 | arn:aws:cloudfront::986325076436:distribution/E3FW6WWZU7BYN5 | C3S3ZT3W3NN10J            | 86400       |
+----------------+--------------------------------------------------------------+---------------------------+-------------+

-  The traffic between the CloudFront distributions and their origins is encrypted

select
  id,
  arn,
  p -> 'CustomOriginConfig' -> 'HTTPPort' as http_port,
  p -> 'CustomOriginConfig' -> 'HTTPSPort' as https_port,
  p -> 'CustomOriginConfig' -> 'OriginKeepaliveTimeout' as origin_keepalive_timeout,
  p -> 'CustomOriginConfig' -> 'OriginProtocolPolicy' as origin_protocol_policy
from
  aws_cloudfront_distribution,
  jsonb_array_elements(origins) as p
where
  p -> 'CustomOriginConfig' ->> 'OriginProtocolPolicy' = 'https-only';

+----------------+--------------------------------------------------------------+-----------+------------+--------------------------+------------------------+
| id             | arn                                                          | http_port | https_port | origin_keepalive_timeout | origin_protocol_policy |
+----------------+--------------------------------------------------------------+-----------+------------+--------------------------+------------------------+
| E3FW6WWZU7BYN5 | arn:aws:cloudfront::986325076436:distribution/E3FW6WWZU7BYN5 | 80        | 443        | 5                        | "https-only"           |
+----------------+--------------------------------------------------------------+-----------+------------+--------------------------+------------------------+

- CloudFront distributions origins use insecure SSL protocols

select
  id,
  arn,
  p -> 'CustomOriginConfig' -> 'OriginSslProtocols' -> 'Items' as items,
  p -> 'CustomOriginConfig' -> 'OriginSslProtocols' -> 'Quantity' as quantity
from
  aws_cloudfront_distribution,
  jsonb_array_elements(origins) as p
WHERE
  p -> 'CustomOriginConfig' -> 'OriginSslProtocols' -> 'Items' ?& array['SSLv3'];

+----------------+--------------------------------------------------------------+---------------------------------------+----------+
| id             | arn                                                          | items                                 | quantity |
+----------------+--------------------------------------------------------------+---------------------------------------+----------+
| E3FW6WWZU7BYN5 | arn:aws:cloudfront::986325076436:distribution/E3FW6WWZU7BYN5 | ["SSLv3","TLSv1","TLSv1.1","TLSv1.2"] | 4        |
+----------------+--------------------------------------------------------------+---------------------------------------+----------+
```
</details>

### CODE REVIEW

File: aws/table_aws_cloudfront_distribution.go [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudfront"
+	"github.com/turbot/steampipe-plugin-sdk/grpc/proto"
+	"github.com/turbot/steampipe-plugin-sdk/plugin"
+	"github.com/turbot/steampipe-plugin-sdk/plugin/transform"
+)
+
+//// TABLE DEFINITION
+
+func tableAwsCloudfrontDistribution(_ context.Context) *plugin.Table {
```
Comment (khushboo9024) [inline aws/table_aws_cloudfront_distribution.go:15]:
```suggestion
func tableAwsCloudFrontDistribution(_ context.Context) *plugin.Table {
```
Please change wherever this function is used.

File: aws/table_aws_cloudfront_distribution.go [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+			},
+			{
+				Name:        "arn",
+				Description: "The ARN (Amazon Resource Name) for the distribution.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ARN", "Distribution.ARN"),
+			},
+			{
+				Name:        "status",
+				Description: "The current status of the Distribution. When the status is Deployed, the distribution's information is propagated to all CloudFront edge locations.",
```
Comment (khushboo9024) [inline aws/table_aws_cloudfront_distribution.go:42]:
```suggestion
				Description: "The current status of the Distribution.",
```

File: aws/table_aws_cloudfront_distribution.go [hunk 3]
```diff
@@ -0,0 +1,329 @@
...
+				Description: resourceInterfaceDescription("tags"),
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getCloudfrontDistributionTags,
+				Transform:   transform.FromField("Tags.Items").Transform(cloudfrontDistributionTagListToTurbotTags),
+			},
+			{
+				Name:        "akas",
+				Description: resourceInterfaceDescription("akas"),
+				Type:        proto.ColumnType_JSON,
+				Transform:   transform.FromField("ARN", "Distribution.ARN").Transform(arnToAkas),
```
Comment (khushboo9024) [inline aws/table_aws_cloudfront_distribution.go:216]: 'title` column is missing.

Comment (khushboo9024) [inline aws/table_aws_cloudfront_distribution.go:216]: 'title` column is missing.

File: docs/tables/aws_cloudfront_distribution.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+select
+  id,
+  arn,
+  status,
+  domain_name,
+  enabled,
+  e_tag,
+  http_version,
+  is_ipv6_enabled
+
```
Comment (khushboo9024) [inline docs/tables/aws_cloudfront_distribution.md:19]:
```suggestion
```

File: docs/tables/aws_cloudfront_distribution.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+
+### List distributions with logging not enabled
+
+```sql
+select
+  id,
+  logging ->> 'Bucket' as bucket,
+  logging ->> 'Enabled' as logging_enable,
```
Comment (khushboo9024) [inline docs/tables/aws_cloudfront_distribution.md:31]:
```suggestion
  logging ->> 'Enabled' as logging_enabled,
```

**Review by khushboo9024 — APPROVED** [review]
LGTM

**Review by rajmohanty17 — CHANGES_REQUESTED** [review]
@visiit @khushboo9024 I do not see we are capturing any encryption info, pls check if we can accommodate below 

- Ensure the traffic between the AWS CloudFront distributions and their origins is encrypted.
- Ensure that Amazon CloudFront web distributions enforce field-level encryption.
- Ensure AWS CloudFront distributions origin(s) do not use insecure SSL protocols.
Ask for more details if you need navigation

File: docs/tables/aws_cloudfront_distribution.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+```sql
+select
+  id,
+  arn,
+  p -> 'CustomOriginConfig' -> 'OriginSslProtocols' -> 'Items' as items,
+  p -> 'CustomOriginConfig' -> 'OriginSslProtocols' -> 'Quantity' as quantity
+from
+  aws_cloudfront_distribution,
+  jsonb_array_elements(origins) as p
+WHERE
```
Comment (rajmohanty17) [inline docs/tables/aws_cloudfront_distribution.md:98]: make it where

File: docs/tables/aws_cloudfront_distribution.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+
+```sql
+select
+  id,
+  arn,
+  default_cache_behavior ->> 'FieldLevelEncryptionId' as field_level_encryption_id,
+  default_cache_behavior ->> 'DefaultTTL' as default_ttl
+from
+  aws_cloudfront_distribution
+WHERE
```
Comment (rajmohanty17) [inline docs/tables/aws_cloudfront_distribution.md:64]: make it where in small case

**Review by rajmohanty17 — CHANGES_REQUESTED** [review]
Small changes required in the examples to lowercase

**Review by rajmohanty17 — APPROVED** [review]
LGTM

**Review by cbruno10 — APPROVED** [review]

**Pull request merged**